examples/ecs/generic_systems.rs

```diff
@@ -1,20 +1,26 @@
 use bevy::prelude::*;
 use bevy::ecs::component::Component;
 
-// When entities spawn in call an event
-
 struct A;
 struct B;
 
-fn spawn_entities<T: Component>(mut commands: Commands, q: Query<Entity, With<T>>) {
-    q.for_each(|entity| { commands.entity(entity).spawn();});
-    info!("Entity {:?} spawned in!")
+pub fn spawn_entities<T: Component>(mut commands: Commands,
+    mouse_button_inputs: Res<Input<MouseButton>>,) {
+    for _ in 0..10 {
+        let id = commands.spawn().insert(T).id();
+        if !mouse_button_inputs.just_pressed(MouseButton::Left) {
+            return info!("Spawned entity {:?} with component {}", id, std::any::type_name::<T>());
+        }
+        if !mouse_button_inputs.just_pressed(MouseButton::Right) {
```

Comment:
These two cases seem identical?

Reply:
What I want to do is when the left or right mouse click is pressed it spawns an entity that's a different color.
For example, left-click spawns a blue entity and right-click spawns a red entity.
Would that still be acceptable for a generic system?

Reply:
@alice-i-cecile One uses `MouseButton::Left` and the other uses `MouseButton::Right`

Reply:
Ah, got it!